heroic-elasticsearch-utils/src/test/java/com/spotify/heroic/elasticsearch/index/ScrollTransformTest.java

```diff
@@ -0,0 +1,340 @@
+package com.spotify.heroic.elasticsearch.index;
+
+import com.spotify.heroic.elasticsearch.AbstractElasticsearchMetadataBackend;
+import com.spotify.heroic.common.ValueOptionalLimit;
+
+import eu.toolchain.async.*;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.OngoingStubbing;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.*;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class ScrollTransformTest {
+    @Mock
+    private AsyncFramework async;
+
+    @Mock
+    private AsyncFuture<AbstractElasticsearchMetadataBackend.LimitedSet> resolved;
+
+    private final SearchHit[] searchHits1 = {
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+    };
+
+    private final SearchHit[] searchHits2 = {
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+        mock(SearchHit.class),
+    };
+
+    private final SearchHit[] emptySearchHits = {};
+
+    @Before
+    public void setup(){
+        when(async.resolved(any(AbstractElasticsearchMetadataBackend.LimitedSet.class))).thenReturn(resolved);
+    }
+
+    public Supplier<AsyncFuture<SearchResponse>> createScroller(SearchHit[]... pages){
+        final SearchResponse searchResponse = mock(SearchResponse.class);
+        final SearchHits searchHits = mock(SearchHits.class);
+        when(searchResponse.getHits()).thenReturn(searchHits);
+        OngoingStubbing<SearchHit[]> stub = when(searchHits.getHits());
+        for(final SearchHit[] page: pages){
+            stub = stub.thenReturn(page);
+        }
+
+        Supplier<AsyncFuture<SearchResponse>> scroller = (Supplier<AsyncFuture<SearchResponse>>) mock(Supplier.class);
+        when(scroller.get()).thenReturn(new AsyncFuture<SearchResponse>() {
```

Comment:
Mock `AsyncFuture<SearchResponse>` with:
```
@Mock
AsyncFuture<SearchResponse> response;
```

And in `@Before`:
```
doAnswer(...).when(response).transform(...)
```
See: https://testing.googleblog.com/2014/03/whenhow-to-use-mockito-answer.html